server/src/main/java/io/crate/expression/reference/sys/snapshot/SysSnapshots.java

```diff
@@ -135,4 +153,22 @@ private static CompletableFuture<SysSnapshot> createSysSnapshot(Repository repos
                 throw Exceptions.toRuntimeException(err);
             });
     }
+
+    /**
+     * Returns a list of currently running snapshots from repository sorted by snapshot creation date
+     *
+     * @param snapshotsInProgress snapshots in progress in the cluster state
+     * @param repositoryName repository name
+     * @return list of snapshots
+     */
+    public static List<SnapshotInfo> sortedCurrentSnapshots(@Nullable SnapshotsInProgress snapshotsInProgress, String repositoryName) {
+        List<SnapshotInfo> snapshotList = new ArrayList<>();
+        List<SnapshotsInProgress.Entry> entries =
+            SnapshotsService.currentSnapshots(snapshotsInProgress, repositoryName, Collections.emptyList());
+        for (SnapshotsInProgress.Entry entry : entries) {
+            snapshotList.add(new SnapshotInfo(entry));
```

Comment:
pushed an update.
Regarding sorting - I think we don't even need it - it was there solely because of the code re-using.

Add a ctor to create `SysSnapshot` directly from `SnapshotInProgress.Entry` to avoid intermediate `SnapshotInfo`/intermediate list creaiton.

Please note, that I utilize `entry.templates` now - so fix for empty parted tables https://github.com/crate/crate/commit/76886ba7159b194e7cf6cdf396b9f81a0fe2e84 kinda automatically backed in into in progress snapshots now